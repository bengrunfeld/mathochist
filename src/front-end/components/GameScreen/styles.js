import styled from 'styled-components'

const Screen = styled.div`
  min-height: 73vh;
`

export { Screen }
