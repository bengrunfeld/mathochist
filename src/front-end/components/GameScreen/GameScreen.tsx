import React from 'react'
import PropTypes from 'prop-types'

import GameInfo from '../GameInfo'
import Hint from '../Hint'
import Problem from '../Problem'
import SolutionInputBox from '../SolutionInputBox'
import ResultDisplay from '../ResultDisplay'

import { Screen } from './styles'

const GameScreen = ({ width }) => (
  <Screen className={width}>
    <GameInfo />
    <Hint />
    <Problem />
    <SolutionInputBox />
    <ResultDisplay />
  </Screen>
)

GameScreen.propTypes = {
  width: PropTypes.string
}

export default GameScreen
