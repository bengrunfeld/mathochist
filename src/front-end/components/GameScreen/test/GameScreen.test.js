import React from 'react'
import Enzyme, { shallow, mount, render } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'

import GameScreen from '../GameScreen'

describe('<GameScreen />', () => {
  test('matches the snapshot', () => {
    const tree = mount(<GameScreen />)
    expect(tree).toMatchSnapshot()
  })
})
