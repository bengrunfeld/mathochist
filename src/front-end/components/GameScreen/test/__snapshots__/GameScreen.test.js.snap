// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameScreen /> matches the snapshot 1`] = `
<GameScreen>
  <styled.div
    className="col-md-6"
  >
    <StyledComponent
      className="col-md-6"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "OGbzu",
            "rules": Array [
              "
  min-height: 73vh;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="col-md-6 sc-bdVaJa OGbzu"
      >
        <GameInfo>
          <div>
            <h1>
              GameInfo
            </h1>
          </div>
        </GameInfo>
        <Hint>
          <div>
            <h1>
              Hint
            </h1>
          </div>
        </Hint>
        <ProblemBox>
          <div>
            <h1>
              ProblemBox
            </h1>
          </div>
        </ProblemBox>
        <SolutionInputBox>
          <div>
            <h1>
              SolutionInputBox
            </h1>
          </div>
        </SolutionInputBox>
        <ResultDisplay>
          <div>
            <h1>
              ResultDisplay
            </h1>
          </div>
        </ResultDisplay>
      </div>
    </StyledComponent>
  </styled.div>
</GameScreen>
`;
